beacon_node/beacon_chain/src/test_utils.rs

```diff
@@ -415,14 +444,68 @@ where
 
         let signed_block = block.sign(
             &self.validator_keypairs[proposer_index].sk,
-            &state.fork,
-            state.genesis_validators_root,
+            &state.fork(),
+            state.genesis_validators_root(),
             &self.spec,
         );
 
         (signed_block, state)
     }
 
+    /// Useful for the `per_block_processing` tests. Creates a block, and returns the state after
+    /// caches are built but before the generated block is processed.
+    pub fn make_block_return_original_state(
```

Comment:
We use `pre_state` to refer to this state in other parts of the code (it also seems like a common term outside of LH too) so perhaps `make_block_return_pre_state` would be more consistent? Just a nit really.

Reply:
Fixed in https://github.com/sigp/lighthouse/pull/2279/commits/6415962290a54c00289cb526aefc5a304e9e94e4